framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4622,450 @@ private void makeGroundTargetType(
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+    if (capturedTypeMirror.getTypeArguments().isEmpty()) {
+      return false;
+    }
+
+    if (uncapturedType.isUnderlyingTypeRaw() || uncapturedType.containsUninferredTypeArguments()) {
+      return false;
+    }
+
+    if (capturedTypeMirror.getTypeArguments().size() != uncapturedType.getTypeArguments().size()) {
+      throw new BugInCF(
+          "Not the same number of type arguments: capturedTypeMirror: %s uncapturedType: %s",
+          capturedTypeMirror, uncapturedType);
+    }
+
+    for (int i = 0; i < capturedTypeMirror.getTypeArguments().size(); i++) {
+      TypeMirror capturedTypeArgTM = capturedTypeMirror.getTypeArguments().get(i);
+      AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+      if (uncapturedTypeArg.getKind() == TypeKind.WILDCARD
+          && (TypesUtils.isCapturedTypeVariable(capturedTypeArgTM)
+              || capturedTypeArgTM.getKind() != TypeKind.WILDCARD)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Apply capture conversion to {@code typeToCapture}.
+   *
+   * <p>Capture conversion is the process of converting wildcards in a parameterized type to fresh
+   * type variables. See <a
+   * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS 5.1.10</a>
+   * for details.
+   *
+   * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+   * this method returns {@code type}.
+   *
+   * @param typeToCapture type to capture
+   * @return the result of applying capture conversion to {@code typeToCapture}
+   */
+  public AnnotatedTypeMirror applyCaptureConversion(AnnotatedTypeMirror typeToCapture) {
+    TypeMirror capturedTypeMirror = types.capture(typeToCapture.getUnderlyingType());
+    return applyCaptureConversion(typeToCapture, capturedTypeMirror);
+  }
+
+  /**
+   * Apply capture conversion to {@code type}.
+   *
+   * <p>Capture conversion is the process of converting wildcards in a parameterized type to fresh
+   * type variables. See <a
+   * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS 5.1.10</a>
+   * for details.
+   *
+   * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+   * this method returns {@code type}.
+   *
+   * @param type type to capture
+   * @param typeMirror the result of applying capture conversion to the underlying type of {@code
+   *     type}; it is used as the underlying type of the returned type
+   * @return the result of applying capture conversion to {@code type}
+   */
+  public AnnotatedTypeMirror applyCaptureConversion(
+      AnnotatedTypeMirror type, TypeMirror typeMirror) {
+
+    // If capture conversion the type contains uninferred type arguments, don't capture, but
+    // mark all wildcards in type as "uninferred" before it is returned.
```

Comment:
This comment says "all wildcards", but there is `&&` after `uncapturedTypeArg.getKind() == TypeKind.WILDCARD`.  Which is right, and why?  (When does the second conjunct have an effect?  Or should it be an assertion?)